### DISCUSSION TITLE
Support for endpoint validation in Azure Functions

### DISCUSSION THREAD

**Discussion by tomkerkhove** [arcus-azure/arcus.eventgrid#173] [Ideas]
Support for endpoint validation in Azure Functions is something you still have to build yourself and did it a few times recently.

It requires the following:
https://github.com/tomkerkhove/autoscaling-over-time-dashboard/blob/main/src/AzureAutoscalingToolbox.Samples.StatefulAppInstances/Functions/AzureMonitorScaledFunction.cs#L29-L32

Not sure if this should be here or a template though, maybe both?

**Comment by stijnmoreels** [comment]
There is other functionality (ex: request tracking) that we have in the web API repo that's also probably valuable for the Azure Functions (templates), but isn't available due to the missing filters/middleware/... For this, we may option to have an extension on the `HttpRequest` that we make available through this repo. That we, we still place it under 'Event Grid' but not directly in our templates? It could be an option though, in the templates, with no relation to this repo. Or both.

**Reply by tomkerkhove** [reply]
Or we can provide a type that can be injected as following:

```csharp
public class AzureMonitorScaledFunction : AzureEventGridFunction
    {
        private readonly IEventGridSubscriptionValidator _eventGridSubscriptionValidator

        public AzureMonitorScaledFunction (IEventGridSubscriptionValidator eventGridSubscriptionValidator)
        {
            _eventGridSubscriptionValidator = eventGridSubscriptionValidator;
        }

        [FunctionName("azure-monitor-scaled-app-event")]
        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "post", "options", Route = "v1/autoscale/azure-monitor/app")] HttpRequest request,
            [DurableClient] IDurableEntityClient durableEntityClient)
        {
            // Option #1: Basic
            _eventGridSubscriptionValidator.ValidateSubscriptionHandshakeIfRequired(request);

            // Option #2: Verbose
            if (request.Method.Equals("OPTIONS", StringComparison.InvariantCultureIgnoreCase))
            {
                return _eventGridSubscriptionValidator.Validate(request);
            }

            // More logic here
        }

}
```

**Reply by stijnmoreels** [reply]
Done! #199
